obspy/io/dirtree/sds.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+"""
+obspy.io.dirtree.sds - read support for SeisComP Data Structure
+===============================================================
+This module provides read support for data stored locally in a SeisComP Data
+Structure (SDS) directory structure.
+
+The directory and file layout of SDS is defined as:
+
+    <SDSdir>/YEAR/NET/STA/CHAN.TYPE/NET.STA.LOC.CHAN.TYPE.YEAR.DAY
+
+These fields are defined by SDS as follows:
+
+    SDSdir :  arbitrary base directory
+    YEAR   :  4 digit year
+    NET    :  Network code/identifier, up to 8 characters, no spaces
+    STA    :  Station code/identifier, up to 8 characters, no spaces
+    CHAN   :  Channel code/identifier, up to 8 characters, no spaces
+    TYPE   :  1 characters indicating the data type, recommended types are:
+               'D' - Waveform data
+               'E' - Detection data
+               'L' - Log data
+               'T' - Timing data
+               'C' - Calibration data
+               'R' - Response data
+               'O' - Opaque data
+    LOC    :  Location identifier, up to 8 characters, no spaces
+    DAY    :  3 digit day of year, padded with zeros
+
+See https://www.seiscomp3.org/wiki/doc/applications/slarchive/SDS.
+
+:copyright:
+    The ObsPy Development Team (devs@obspy.org)
+:license:
+    GNU Lesser General Public License, Version 3
+    (http://www.gnu.org/copyleft/lesser.html)
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+
+import glob
+import os
+from datetime import timedelta
+
+from obspy import Stream, read
+from obspy.core.util.misc import BAND_CODE
+
+
+SDS_FMTSTR = os.path.join(
+    "{year}", "{network}", "{station}", "{channel}.{type}",
+    "{network}.{station}.{location}.{channel}.{type}.{year}.{doy:03d}")
```

Comment:
This probably complicates things a bit but I think it should be possible to make this configurable. 

Instead of `doy` I would also use `julday`. Sounds less like a typo and is what we use elsewhere.